File: generator/src/main/java/ch/bailu/gtk/builder/BuilderInterface.java
package ch.bailu.gtk.builder;

import java.io.IOException;

import ch.bailu.gtk.tag.AliasTag;
import ch.bailu.gtk.tag.CallbackTag;
import ch.bailu.gtk.tag.EnumerationTag;
import ch.bailu.gtk.tag.NamespaceTag;
import ch.bailu.gtk.tag.StructureTag;

public interface BuilderInterface {
    void buildStructure(StructureTag structure) throws IOException;
    void buildNamespaceStart(NamespaceTag namespace);
    void buildNamespaceEnd(NamespaceTag namespace) throws IOException;
    void buildAlias(AliasTag alias);
    void buildEnumeration(EnumerationTag enumeration) throws IOException;
    void buildCallback(CallbackTag callbackTag) throws IOException;
}

File: generator/src/main/kotlin/ch/bailu/gtk/builder/BuilderInterface.kt
package ch.bailu.gtk.builder

import ch.bailu.gtk.parser.tag.*
import ch.bailu.gtk.config.NamespaceConfig
import java.io.IOException

interface BuilderInterface {
    @Throws(IOException::class)
    fun buildStructure(structure: StructureTag)
    fun buildNamespaceStart(namespace: NamespaceTag, namespaceConfig: NamespaceConfig)

    @Throws(IOException::class)
    fun buildNamespaceEnd()
    fun buildAlias(aliasTag: AliasTag)

    @Throws(IOException::class)
    fun buildEnumeration(enumeration: EnumerationTag)

    @Throws(IOException::class)
    fun buildCallback(callbackTag: CallbackTag)
    
    fun buildErrorStubs(enabled: Boolean)
}
